### ISSUE TITLE
"flicking between metrics" is harder now than it was

### ISSUE THREAD

**Issue by rade** [weaveworks/scope#1226]
I really like how metrics-on-canvas allows me to flick between metrics by simply moving my mouse over the list. The whole thing could be done in one smooth gesture, without paying close attention to how the mouse pointer was moving. *It worked perfectly!*

Not anymore though...

The list used to be vertical, which did two things: 1) movement did not have to be precisely vertical since the list was quite wide, 2) equal spacing meant moving the mouse at constant speed would flick metrics at constant rate. 

The horizontal list doesn't work that way. It is only a few pixels tall and hence requires a lot of precision in mouse movement - more than I can manage without looking. And the items have different width, so the "flick rate" is variable.

@davkal @foot

**Comment by davkal** [comment]
This comes down to tradeoffs:

* wide lists were more forgiving but encroached upon the main canvas, making it look more cluttered
* integrating the metric chooser in the button groups prevented us from introducing yet another UI element which would again have added clutter, but came at a loss of "constant flick rate"

In both points I support the current implementation, so :-1: on the issue. @foot ?

For pros we should advertise the `<`, `>`, etc a bit more. You can flick faster and constant with you keyboard.

**Comment by rade** [comment]
> For pros we should advertise the `<`, `>`, etc a bit more

Huh? Neither the arrow keys nor the less/greater-than keys do anything for me in either FF or Chrome.

Also "a bit more" == "at all"! How was I supposed to know that there are keyboard shortcuts?

**Comment by foot** [comment]
Okay thats not working in FF, interesting.

In Chrome they should be behaving (lt/gt symbols). But discoverability as mentioned is lowww.

**Comment by rade** [comment]
I can assure you this does not work for me even in Chrome. Perhaps it works in Chrome *on a Mac* only.

**Comment by foot** [comment]
Yep, there is a discoverability middle ground with a "drag-back-and-forth":
while the mouse is down look for the closest column: e.g. expand the hover
space out infinitely up and down. This doesn't work well with trackpads,
but with mouse its quite slick.

KB shortcuts do have a lot of potential here. I'm not 100% w/ `<` / `>`.
They do feel a tad awkward. Arrow keys? I like vim :). hjkl ftw.

**Comment by rade** [comment]
>  expand the hover space out infinitely up and down

That would not provide a "constant flick rate", due to the variable width of the infinite bands.

**Comment by davkal** [comment]
I created #1228 for the hotkey issue. 

(1) Regarding this one, the premise is a bit off. The observed "perfect behavior" was based on a prototype that aimed for quick integration w/o regards to how it fits into the rest of the product.

(2) The variable flick rate and small hover space are valid observations and a potential usability issue.
But I cant see how this can be solved without redesigning the lower left corner, making it a bigger issue.

Based on (1) I would close this issue. But if (2) gets more support we could keep this open. @foot @rade?

**Comment by rade** [comment]
"flicking" is an important aspect of MoC; it is *the* way to quickly discover whether/where there is "something unusual". And the visual focus must be on the nodes for that to work, i.e. the flicking must be possible without shifting focus.

I don't care how that is achieved; I am not wedded to the vertical list and mouse gesture.

**Comment by rade** [comment]
incidentally, as you say, keyboard shortcuts would work better than mouse gestures. Except for their discoverability, and the fact that the mouse is the main input device for navigating scope, so switching to keyboard interaction requires a extra effort.

**Comment by davkal** [comment]
> and the fact that the mouse is the main input device for navigating scope, so switching to keyboard interaction requires a extra effort.

That ties in well with my last comment in #1228. I think there is potential to be more keyboard driven especially for very big topologies.
